src/tests/with-effect.ts

```diff
@@ -0,0 +1,26 @@
+import { Component } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+
+/**
+ * Testing function which executes inside of an injectionContext and provides Change Detection trigger.
+ *
+ * It looks like, we need to have at least one component to execute effects.
+ * AppRef::tick() does not work. Only ComponentFixture::detectChanges() seems to do the job.
+ * withEffects renders a TestComponent and executes the tests inside an injectionContext.
+ * This allows us to generate effects very easily.
+ */
+export const withEffect =
```

Comment:
I'd rename this function to avoid potential naming confusion with SignalStore features:

```suggestion
export const testEffect =
```

Reply:
done